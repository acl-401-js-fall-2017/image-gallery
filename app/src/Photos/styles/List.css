tr:nth-of-type(2n) {
    background-color: lightgray;
}

table {
    border-bottom: 3px solid black;
    width: 100%;
}

th, td {
    padding: 0.5em 0.5em;
}