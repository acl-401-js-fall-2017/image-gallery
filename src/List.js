import React, { PureComponent } from 'react';
import styled from 'styled-components';
import { loadGallery } from './actions'

export default class List extends PureComponent {
  state ={
    gallery: []
  }

  componentDidMount() {
    const newState = loadGallery(this.state);
    this.setState(newState);
  }

  render() {
    return( 
      <StyledDiv>
        <ul>
          {this.state.gallery.map(img => (
            <div key={img._id} style ={{display: 'flex', margin: '8px'}}>
              <li type= "none">{img.title}: {img.description}</li>
            </div>
          ))}
        </ul>
      </StyledDiv>
    )
  };
}

const StyledDiv = styled.div`
display:'flex';
width: 90%;
justify-content:space-between;
`