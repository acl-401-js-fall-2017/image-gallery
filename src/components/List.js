import React, { Component } from 'react';
import PropTypes from 'prop-types';

import AddImage from './AddImage';

export default class List extends Component {
   
    static propTypes = {
      images: PropTypes.arrayOf(PropTypes.object).isRequired
    }


    render() {
      const { addImage, handleRemove } = this.props;
      return (
        <div className="list-view">
          <table>
            <tbody>
              {this.props.images.map((image, index) => {
                return (
                  <tr key={index}>
                    <td><a href={image.url}>{image.title}</a></td>
                    <td>{image.description}</td>
                    <td> <input type="button" data-value={image._id} value="remove" onClick={() => handleRemove(image._id)}/></td>
                  </tr>
                );
              })}
              <tr>
                <td></td>
                <td><AddImage addImage={newImage => addImage(newImage)}/></td>
                <td></td>
              </tr>
            </tbody>
          </table>
        </div>
      );
    }
}