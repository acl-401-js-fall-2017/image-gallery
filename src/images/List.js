import React, { PureComponent } from 'react';
import styled from 'styled-components';
import { loadGallery, deleteImg } from '../utils/actions';
import imgApi from '../utils/services/imgApi';

export default class List extends PureComponent {
  state ={
    gallery: []
  }

  getAlbumId() {
    return this.props.match.params.id;
  }

  async componentDidMount() {
    const img = await imgApi.get(this.getAlbumId());
    const newState = loadGallery(this.state, img);
    this.setState(newState);
  }

  handleDelete = async id => {
    await imgApi.remove(id);
    const newState = deleteImg(this.state, id);
    this.setState(newState);
  }

  render() {
    return( 
      <StyledDiv>
        <ul>
          {this.state.gallery.map(img => (
            <div key={img._id} style ={{ display: 'flex', margin: '8px' }}>
              <DeleteDiv onClick={() => this.handleDelete(img._id)}>X</DeleteDiv>
              <li type= "none">{img.title}: {img.description}</li>
            </div>
          ))}
        </ul>
      </StyledDiv>
    );
  }
}

const StyledDiv = styled.div`
display:'flex';
width: 90%;
justify-content:space-between;
`;

const DeleteDiv = styled.div`
border: 1px solid black;
margin-right: 1%;
`;