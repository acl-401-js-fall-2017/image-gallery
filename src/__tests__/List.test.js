import React from 'react';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Bunnies from '../images/bunnies';
import List from '../components/List';

describe('List', () => {
  it('lists of bunnies', () => {
    const wrapper = shallow(<List bunnies={Bunnies}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});