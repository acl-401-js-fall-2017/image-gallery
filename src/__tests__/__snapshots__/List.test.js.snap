// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List lists of bunnies 1`] = `
<div
  className="list-view"
>
  <table>
    <tbody>
      <tr
        key="0"
      >
        <td>
          <a
            href="https://s3.amazonaws.com/f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
          >
            Best Bunny
          </a>
        </td>
        <td>
          Just the cutest color on a bunny ever
        </td>
        <td>
           
          <input
            data-value="1"
            onClick={[Function]}
            type="button"
            value="remove"
          />
        </td>
      </tr>
      <tr
        key="1"
      >
        <td>
          <a
            href="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg"
          >
            Kind of cute Bunny
          </a>
        </td>
        <td>
          Too many carbs :(
        </td>
        <td>
           
          <input
            data-value="2"
            onClick={[Function]}
            type="button"
            value="remove"
          />
        </td>
      </tr>
      <tr
        key="2"
      >
        <td>
          <a
            href="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg"
          >
            Tiny Bunny
          </a>
        </td>
        <td>
          Cant see me
        </td>
        <td>
           
          <input
            data-value="3"
            onClick={[Function]}
            type="button"
            value="remove"
          />
        </td>
      </tr>
      <tr>
        <td />
        <td>
          <AddImage
            addImage={[Function]}
          />
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
